File: exam-system-java/entity/src/main/java/com/yyjy/exam/entity/question/entity/QuestionChoices.java
package com.yyjy.exam.entity.question.entity;

import org.babyfish.jimmer.sql.*;
import org.jetbrains.annotations.Nullable;

import java.time.LocalDateTime;

@Entity
public interface QuestionChoices {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    Questions question();

    @IdView
    long questionId();

    String content();

    @Nullable
    @Column(name = "is_correct")
    Boolean correct();

    @Nullable
    Integer sort();

    @Nullable
    LocalDateTime createTime();

    @Nullable
    LocalDateTime updateTime();

    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    @Nullable
    LocalDateTime delFlag();
}

File: exam-system-server/src/main/kotlin/com/yyjy/models/entity/QuestionChoices.kt
package com.yyjy.models.entity

import org.babyfish.jimmer.sql.*
import java.time.LocalDateTime

/**
 * 题目选项表（选择题选项）
 */
@Entity
interface

QuestionChoices {

    @Id
    @GeneratedValue(
        strategy = GenerationType.IDENTITY
    )
    val id: Long

    @OnDissociate(DissociateAction.DELETE)
    @ManyToOne
    val question: Questions

    @IdView
    val questionId: Long

    val content: String

    @Column(name = "is_correct")
    val correct: Boolean?

    val sort: Int?

    /**
     * 创建时间
     */
    val createTime: LocalDateTime?

    /**
     * 更新时间
     */
    val updateTime: LocalDateTime?

    /**
     * 删除时间
     */
    @Column(name = "deleted_at")
    @LogicalDeleted("now")
    val delFlag: LocalDateTime?
}

